src/sentry/similarity/index.py

```diff
@@ -60,8 +60,12 @@ def __build_signatures(self, items):
 
         return arguments
 
-    def _get_connection(self, scope):
-        return self.cluster.get_local_client_for_key(scope)
+    def __index(self, scope, args):
+        # scope must be passed into the script call as a key to allow the
+        # cluster client to determine what cluster the script should be
+        # executed on. The script itself will use the scope as the hashtag for
+        # all redis operations.
+        return index(self.cluster, [scope], args)
```

Comment:
I didn't verify this runs correctly locally against a cluster, I'm assuming you checked this out (since the tests use a vanilla Redis connection.) It'd be nice to have the tests run against an actual big-C Redis Cluster in the future.